store/root/store.go

```diff
@@ -255,6 +268,21 @@ func (s *Store) WorkingHash(cs *store.Changeset) ([]byte, error) {
 	}
 
 	if s.workingHash == nil {
+		// if migration is in progress, send the changeset to the migration manager
+		if s.isMigrating {
```

Comment:
This seems like an odd place to be doing synchronization logic about migrations, and seems to hinge on this arbitrary constraint in the comment:

> // WorkingHash returns the working hash of the root store. Note, WorkingHash()
> // should only be called once per block once all writes are complete and prior
> // to Commit() being called.

Why should WorkingHash only be called once?

Reply:
This is a great question. Technically speaking, if you look at `root.Store#WorkingHash`, it's actually idempotent. So technically, it can be called multiple times.

The comment is mainly to infer expected workflow: ABCI.FinalizeBlock -> RootStore.WorkingHash() -> ABCI.Commit(). This doesn't mean it can't be called twice, but rather, it's only expected to be called once.

Regarding placement, why do we hinge or rely on `if s.workingHash == nil` @cool-develope ?